mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -6724,23 +6764,39 @@ struct TransformRewritePattern : public OpRewritePattern<miopen::TransformOp> {
                   AffineMapAttr::get(outputType.getAffineMaps()[0]));
 
               auto existingDomain = dictAttr.get("domain").cast<ArrayAttr>();
+              auto existingLimits = dictAttr.get("limits");
 
-              augmentedArrayAttr.push_back(b.getDictionaryAttr(
-                  {b.getNamedAttr("operand",
-                                  b.getI32IntegerAttr(userOperandIndex)),
-                   b.getNamedAttr("transforms",
-                                  b.getArrayAttr(augmentedTransforms)),
-                   b.getNamedAttr("domain", existingDomain)}));
+              llvm::SmallVector<NamedAttribute, 4> arrayAttr{
+                  b.getNamedAttr("operand",
+                                 b.getI32IntegerAttr(userOperandIndex)),
+                  b.getNamedAttr("transforms",
+                                 b.getArrayAttr(augmentedTransforms)),
+                  b.getNamedAttr("domain", existingDomain)};
+
+              if (existingLimits)
+                arrayAttr.push_back(
+                    b.getNamedAttr("limits", existingLimits.cast<ArrayAttr>()));
+
+              augmentedArrayAttr.push_back(
+                  b.getDictionaryAttr({arrayAttr.begin(), arrayAttr.end()}));
               augmented = true;
             }
           }
-          if (!augmented)
-            augmentedArrayAttr.push_back(b.getDictionaryAttr(
-                {b.getNamedAttr("operand",
-                                b.getI32IntegerAttr(userOperandIndex)),
-                 b.getNamedAttr("transforms", b.getAffineMapArrayAttr(
-                                                  outputType.getAffineMaps())),
-                 b.getNamedAttr("domain", b.getArrayAttr(shapeAttrVec))}));
+          if (!augmented) {
+            llvm::SmallVector<NamedAttribute, 4> arrayAttr{
+                b.getNamedAttr("operand",
+                               b.getI32IntegerAttr(userOperandIndex)),
+                b.getNamedAttr(
+                    "transforms",
+                    b.getAffineMapArrayAttr(outputType.getAffineMaps()[0])),
```

Comment:
same here. we need to use comment to explain what `[0]` and `[1]` are.

Reply:
done